firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/Chunks.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ * Chunks.cpp
+ *
+ *  Created on: Nov 3, 2020
+ *      Author: Renji Panicker
+ *
+ * Unit tests to confirm behavior of chunk splitter
+ *
+ */
+
+#include "Pufferfish/Protocols/Chunks.h"
+
+#include "Pufferfish/Util/Vector.h"
+#include "catch2/catch.hpp"
+
+namespace PF = Pufferfish;
+
+SCENARIO(
+    "Protocols::ChunkSplitter behaves correctly across chunk boundary(256 bytes)", "[chunks]") {
+  GIVEN("A chunk splitter") {
+    constexpr size_t buffer_size = 256;
+    PF::Protocols::ChunkSplitter<buffer_size, uint8_t> chunks;
+
+    WHEN("the input data exceeds bounds after output is called, without delimiter") {
+      uint8_t val = 128;
+      PF::Protocols::ChunkInputStatus status;
+      for (size_t i = 0; i < buffer_size; ++i) {
+        status = chunks.input(val);
+      }
+      PF::Util::Vector<uint8_t, buffer_size> buffer;
+      PF::Protocols::ChunkOutputStatus output_status = chunks.output(buffer);
+      auto final_status = chunks.input(val);
+
+      THEN("the final status should be ok") {
+        REQUIRE(status == PF::Protocols::ChunkInputStatus::ok);
+        REQUIRE(output_status == PF::Protocols::ChunkOutputStatus::waiting);
+        REQUIRE(final_status == PF::Protocols::ChunkInputStatus::invalid_length);
+      }
+    }
+  }
+}
+
+SCENARIO(
+    "Protocols::ChunkSplitter behaves correctly across chunk boundary(102 bytes)", "[chunks]") {
```

Comment:
Was there any specific reason to use 102 as the buffer size for most of the tests and 256 as the buffer size for only one test, rather than the other way around? I think the buffer size we use for most of our tests should be 256, since that's what we use in the actual code (see https://github.com/pez-globo/pufferfish-software/blob/develop/firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/Serial/Backend/Frames.h)

Reply:
Changed to 256 for most cases, and 102 for one.